bin/bb/cmd/migrate.go

```diff
@@ -10,106 +10,62 @@ import (
 	"strings"
 	"time"
 
+	"github.com/bytebase/bytebase/bin/bb/cmdutils"
 	"github.com/bytebase/bytebase/plugin/db"
+	"go.uber.org/zap"
 
 	// install mysql driver.
 	_ "github.com/bytebase/bytebase/plugin/db/mysql"
 	"github.com/spf13/cobra"
 )
 
-func newMigrateCmd() *cobra.Command {
+func newMigrateCmd(ctx context.Context, logger *zap.Logger) *cobra.Command {
 	var (
-		databaseType string
-		username     string
-		password     string
-		hostname     string
-		port         string
-		database     string
-		fileList     []string
-		commandList  []string
-		description  string
-		issueID      string
-
-		// SSL flags.
-		sslCA   string // server-ca.pem
-		sslCert string // client-cert.pem
-		sslKey  string // client-key.pem
+		fileList    []string
+		commandList []string
+		description string
+		issueID     string
 	)
 	migrateCmd := &cobra.Command{
 		Use:   "migrate",
 		Short: "Migrate the database schema",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			tlsCfg := db.TLSConfig{
-				SslCA:   sslCA,
-				SslCert: sslCert,
-				SslKey:  sslKey,
-			}
-
-			var sqlReaders []io.Reader
-
-			//TODO(qsliu): support file and command combined as the passed order.
-			for _, file := range fileList {
-				f, err := os.Open(file)
-				if err != nil {
-					return err
-				}
-				defer f.Close()
-				sqlReaders = append(sqlReaders, f)
-			}
-
-			for _, command := range commandList {
-				sqlReaders = append(sqlReaders, strings.NewReader(command))
-			}
-
-			sqlReader := io.MultiReader(sqlReaders...)
-			return migrateDatabase(context.Background(), databaseType, username, password, hostname, port, database, description, issueID, false /*createDatabase*/, sqlReader, tlsCfg)
-		}}
-
-	migrateCmd.Flags().StringVar(&databaseType, "type", "mysql", "Database type. (mysql or pg).")
-	migrateCmd.Flags().StringVar(&username, "username", "", "Database username. (default mysql:root pg:postgres).")
-	migrateCmd.Flags().StringVar(&password, "password", "", "Database password.")
-	migrateCmd.Flags().StringVar(&hostname, "host", "", "Database host.")
-	migrateCmd.Flags().StringVar(&port, "port", "", "Port of database. (default mysql:3306 pg:5432).")
-	migrateCmd.Flags().StringVar(&database, "database", "", "Target database to execute migration.")
+	}
+	dbOption := cmdutils.NeedDatabaseDriver(migrateCmd)
 	migrateCmd.Flags().StringSliceVarP(&fileList, "file", "f", []string{}, "SQL file to execute.")
```

Comment:
If the existing tool does not support the list, I would suggest we also not support them for now.

A feature is also a liability.

Reply:
This feature comes from psql, see https://www.postgresql.org/docs/13/app-psql.html#:~:text=This%20option%20can%20be%20repeated%20and%20combined%20in%20any%20order%20with%20the%20%2Df%20option.

I think supporting single file or single command is okay. The only question is how do we do when -f and -c passed together?

Reply:
I suggest not to support it now. Let's add it when necessary. -c and -f should be mutually exclusive.

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
this feature is reverted. now ignore -c when both are specified